lang/py/avro/io.py

```diff
@@ -362,25 +362,52 @@ def read_time_micros_from_long(self) -> datetime.time:
         microseconds = self.read_long()
         return self._build_time_object(microseconds, 1)
 
-    def read_timestamp_millis_from_long(self) -> datetime.datetime:
+    def _read_timestamp_millis_from_long(self, tz:bool) -> datetime.datetime:
         """
         long is decoded as python datetime object which represents
         the number of milliseconds from the unix epoch, 1 January 1970.
         """
         timestamp_millis = self.read_long()
         timedelta = datetime.timedelta(microseconds=timestamp_millis * 1000)
-        unix_epoch_datetime = datetime.datetime(1970, 1, 1, 0, 0, 0, 0, tzinfo=avro.timezones.utc)
+        unix_epoch_datetime = datetime.datetime(1970, 1, 1, 0, 0, 0, 0, tzinfo=avro.timezones.utc if tz else None)
         return unix_epoch_datetime + timedelta
 
-    def read_timestamp_micros_from_long(self) -> datetime.datetime:
+    def _read_timestamp_micros_from_long(self,tz:bool) -> datetime.datetime:
         """
         long is decoded as python datetime object which represents
         the number of microseconds from the unix epoch, 1 January 1970.
         """
         timestamp_micros = self.read_long()
         timedelta = datetime.timedelta(microseconds=timestamp_micros)
-        unix_epoch_datetime = datetime.datetime(1970, 1, 1, 0, 0, 0, 0, tzinfo=avro.timezones.utc)
+        unix_epoch_datetime = datetime.datetime(1970, 1, 1, 0, 0, 0, 0, tzinfo=avro.timezones.utc if tz else None)
         return unix_epoch_datetime + timedelta
+    def read_local_timestamp_micros_from_long(self) -> datetime.datetime:
```

Comment:
The way I read the Java implementation, the actual long is still the ms or us from the utc epoch time, however th returned datetime is converted to local time zone.

```
 def read_local_timestamp_millis_from_long(self) -> datetime.datetime:
        return self.read_timestamp_millis_from_long().astimezone()

 def read_local_timestamp_micros_from_long(self) -> datetime.datetime:
        return self.read_timestamp_micros_from_long().astimezone()
```

since `astimezone()` converts to local tz. And the original `read_timestamp_micros/millis_from_long` functions can stay the same.

Reply:
Are you sure about this? I am not great at reading Java, but I understand that a `LocalDateTime` is essentially a DateTime without a timezone, which is what I'm doing here: setting `tz=None` if the type is local.